### PR TITLE
fix: defer calculation of calc answers until after creating a new doc

### PR DESCRIPTION
**Pull request by open-dynaMIX**

### CODE REVIEW

File: caluma/caluma_form/api.py [hunk 1]
```diff
@@ -92,3 +93,62 @@ def copy_form(
...
+    )
+
+
+def recalculate_answers_from_document(instance):
+    if instance.family.meta.get("_defer_calculation"):
+        return
+    for question in models.Form.get_all_questions(
+        [(instance.family or instance).form_id]
+    ).filter(type=models.Question.TYPE_CALCULATED_FLOAT):
+        update_or_create_calc_answer(question, instance)
```
Comment (winged) [inline caluma/caluma_form/api.py:154]: I think we should split the PR into the refactoring bit and the actual bugfix bit, so make it two commits. That way, we won't have any confusion when we're digging for bugs in the future

Also, I think we don't really need to move the `update_calc_dependents()` function, thus avoiding the circular import here.

**Review by winged — CHANGES_REQUESTED** [review]
LGTM, was a nice pairing session! I'd split out the commits as noted tho

File: caluma/caluma_form/api.py (hunk 1 above)
Comment (open-dynaMIX) [reply caluma/caluma_form/api.py:154]: > Also, I think we don't really need to move the update_calc_dependents() function, thus avoiding the circular import here.

The problem isn't `update_calc_dependents()`, but the import from `caluma.caluma_form.api` inside `caluma.caluma_form.domain_logic`. I've solved it with a delayed import. Will split the commit now and the make CI pass.

**Review by winged — APPROVED** [review]
:rocket:

**Pull request merged**